optimizer/resolver.go

```diff
@@ -638,6 +639,7 @@ func (nr *nameResolver) resolveColumnInResultFields(ctx *resolverContext, cn *as
 			if rf.Column.Name.L == "" {
 				// This is not a real table column, resolve it directly.
 				cn.Refer = rf
+				rf.Referenced = true
```

Comment:
Does it necessary when you set all resultField Referenced in createResultFields?